src/coreclr/scripts/superpmi.py

```diff
@@ -1804,8 +1824,12 @@ def replay_with_asm_diffs(self):
                         if return_code == 0:
                             logging.info("Clean SuperPMI replay")
                         else:
-                            files_with_replay_failures.append(mch_file)
                             result = False
+                            # Don't report as replay failure asm diffs (return code 2) or missing data (return code 3).
+                            # Anything else, such as compilation failure (return code 1, typically a JIT assert) will be
+                            # reported as a replay failure.
+                            if return_code != 2 and return_code != 3:
```

Comment:
This is superpmi replay, so why would it get `return_code == 2` (asm diffs failure)?

Reply:
This is the asm diffs case (only).

I'll go ahead and add similar "ignore" code to the `replay` case, but only for missing data.